### PR TITLE
Opt in to analysis in new plugin arch via yaml file

### PR DESCRIPTION
**Pull request by MichaelRFairhurst**
Since `ServerPlugin` expects a driver and calls methods on it behind
the scenes, return a `NoopDriver` for the analysis roots that don't
have angular enabled.

Our own code has to detect the NoopDrivers and bail out early, but
that's easy enough to add.

Expects your `analysis_options.yaml` to have:

```
angular:
  enabled: true
```

In order to run our analysis.

New dependency on package:yaml, and handle edge cases like file not
actually existing, etc.

Tests for that as well, with mocks thanks to a little FileService that
gets injected.

### CODE REVIEW

File: angular_analyzer_plugin/lib/src/file_service.dart [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import 'dart:io';
+
+/// Class for testing. This can be mocked to provide mock files.
+class FileService {
```
Comment (bwilkerson) [inline angular_analyzer_plugin/lib/src/file_service.dart]: Is there a reason why you're not using `ResourceProvider` for this purpose?

File: angular_analyzer_plugin/test/plugin_test.dart [hunk 2]
```diff
@@ -17,18 +22,34 @@ void main() {
 class PluginIntegrationTest {
   AngularAnalyzerPlugin plugin;
   ResourceProvider resourceProvider;
+  FileService fileService;
+  protocol.ContextRoot root;
 
   void setUp() {
     resourceProvider = PhysicalResourceProvider.INSTANCE;
```
Comment (bwilkerson) [inline angular_analyzer_plugin/test/plugin_test.dart]: For testing, we usually create a `MemoryResourceProvider`. It allows us to create a file system in memory so we don't need to actually create and destroy files on disk, which makes the tests run much faster.

File: angular_analyzer_plugin/lib/src/file_service.dart (hunk 1 above)
Comment (MichaelRFairhurst) [inline angular_analyzer_plugin/lib/src/file_service.dart]: Nope! Will change.

**Comment by zoechi** [comment]
I'm wondering if

```yml
angular:
  enabled: true
```
is the best idea

The analyzer complains that `angular` is not a valid key

Perhaps something like

```yml
plugins: 
  angular:
    enabled: true
```
can be used where `plugins` is made known to analyzer so that it doesn't complain and ignores child values.
If you have another strategy in mind, please just ignore my comment.

**Comment by bwilkerson** [comment]
I like the idea of putting all of the plug-in specific options under a top-level `plugins` key. It will reduce the change of name conflicts later.

**Comment by MichaelRFairhurst** [comment]
Agreed, the reasoning is solid. Will make the change.

>
